tests/test_forms.py

```diff
@@ -652,6 +652,31 @@ def required_slots(_tracker):
     )
 
 
+def test_validate_prefilled_slot():
+    # noinspection PyAbstractClass
+    class CustomFormAction(FormAction):
+        def name(self):
+            return "some_form"
+
+        @staticmethod
+        def required_slots(_tracker):
+            return ["some_slot", "some_other_slot"]
+
+        def validate_some_slot(self, value, dispatcher, tracker, domain):
+            if value == "some_value":
+                return "validated_value"
+
+    form = CustomFormAction()
+
+    tracker = Tracker(
+        "default", {"some_slot": "some_value"}, {}, [], False, None, {}, "form"
```

Comment:
I think this test overfits to your code. Hardcoding concrete previous action name `form` and then testing for it